Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionCiscoASA.yml

```diff
@@ -0,0 +1,369 @@
+                        EventType = "NetworkSession",
+                        EventSchema = "NetworkSession",
+                        EventSchemaVersion = "0.2.4",
+                        DvcAction = case(trim(@"\s",SimplifiedDeviceAction) in ("Teardown","Built"), "",
+                                        SimplifiedDeviceAction),
+                        SrcInterfaceName = tolower(SrcInterfaceName),
+                        DstInterfaceName = tolower(SrcInterfaceName)
+                | lookup ProtocolLookup on Protocol
+                | project-rename 
+                        EventProductVersion = DeviceVersion,
+                        EventOriginalType = DeviceEventClassID,
+                        EventOriginalSeverity = OriginalLogSeverity,
+                        DvcOriginalAction = DeviceAction,
+                        EventMessage = Message,
+                        Dvc = Computer
+                | extend
+                        EventSeverity = iff(isempty(DvcAction) or DvcAction == "Allow", "Informational", "Low"),                             
+                        DstPortNumber = coalesce(DstPortNumber, DestinationPort),
+                        SrcPortNumber = coalesce(SrcPortNumber, SourcePort),
+                        DstIpAddr = coalesce(DstIpAddr, DestinationIP),
+                        SrcIpAddr = coalesce(SrcIpAddr, SourceIP),
+                        NetworkRuleName = coalesce(NetworkRuleName, DeviceCustomString2),
+                        NetworkDirection = case(isnotempty(CommunicationDirection), CommunicationDirection,
+                                    (SrcInterfaceName in (internalInterface) and DstInterfaceName in (externalInterface)) or DstInterfaceName in (externalInterface), "Outbound",
+                                    (SrcInterfaceName in (externalInterface) and DstInterfaceName in (internalInterface)) or SrcInterfaceName in (externalInterface), "Inbound",
+                                    SrcInterfaceName in (internalInterface) and DstInterfaceName in (internalInterface), "Local",
+                                    ""),
+                        NetworkProtocol = case(isempty(NetworkProtocol) and isnotempty(Protocol), toupper(Protocol),
+                                                ""),
+                        EventResult = case(isnotempty(EventResult), EventResult,
+                                            DvcAction == "Allow", "Success",
+                                            "Failure")
+                | extend 
+                        Src = SrcIpAddr,
+                        Dst = DstIpAddr,
+                        Duration = NetworkDuration,
+                        IpAddr = SrcIpAddr,
+                        Rule = NetworkRuleName,
+                        User = DstUsername
+                | project-away CommunicationDirection, Destination*, Device*, LogSeverity, Protocol, SimplifiedDeviceAction, Source*
```

Comment:
1. The list of fields we project away seems to be shorter than the Checkpoint parser. Just making sure we did no forget any.
2. While i does not look "nice", project-away at each union branch everything you can. It significantly reduces the union memory cost. If you need a specific Device field for example, rename it in the branch so you can projec-away Device*

Reply:
1. The list is smaller here because I have used project-away in allLogs itself.
2. I have used this technique to project-away at the root query itself. Also, any temp variables created in individual branch are also project-away.